genomic_data_service/region_indexer_file_reader.py

```diff
@@ -10,13 +10,17 @@
 MAX_IN_MEMORY_FILE_SIZE = (25 * 1024 * 1024)
 
 
-class S3BedFileRemoteReader():
-    def __init__(self, file_properties, strand_col_values, snp_set=False):
-        self.file_properties = file_properties
+class FileReader():
+    def __init__(self, file_path, strand_col_values={}, file_size=0, source="s3"):
+        self.file_size = file_size
+        self.file_path = file_path
         self.temp_file = tempfile.NamedTemporaryFile()
-        self.snp_set = snp_set
         self.strand_values = strand_col_values
-        self.data = self.download_file_from_s3()
+        self.data = None
+        if source == "s3":
+            self.data = self.download_file_from_s3()
+        elif source == "local":
+            self.data = gzip.open(self.file_path, mode='rt')
 
 
     def close(self):
```

Comment:
I wonder if we could figure out a way to write these classes without needing to use this? It should not be responsibility of the client to close files that are internal to the object they are using.